src/components/display/renderer/ConditionalRenderingWrapper.tsx

```diff
@@ -0,0 +1,14 @@
+import { useFormContext } from 'react-hook-form';
+import { ReactNode } from 'react';
+
+export interface ConditionalRenderingProps {
+  shouldRender?: (values: Record<string, any>) => boolean;
+  children: ReactNode;
+}
+
+export function ConditionalRenderingWrapper({ shouldRender, children }: ConditionalRenderingProps) {
+  const { watch } = useFormContext();
+  const renderChildren: boolean = shouldRender ? shouldRender(watch()) : true;
```

Comment:
Can we optimize this a bit?
```suggestion
  const values = watch()
  const renderChildren: boolean = useMemo(() => shouldRender(values), [values])
```

Reply:
definitely 🤝